### PR TITLE
TRACING-3262 | Document OTEL TLS options

### PR DESCRIPTION
**Pull request by pavolloffay**
Version(s):

4.11
4.12
4.13
4.14

Issue:

https://issues.redhat.com/browse/TRACING-3262

Link to docs preview:

https://64719--docspreview.netlify.app/

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by pavolloffay** [comment]
The PR can be merged ASAP, it's for RHOSDT 2.9

File: modules/distr-tracing-otel-config-collector.adoc [hunk 1]
```diff
@@ -172,8 +175,11 @@ The OTLP receiver ingests data using the OpenTelemetry protocol (OTLP).
           receivers: [otlp]
 ----
 <1> The OTLP gRPC endpoint. If omitted, the default `+0.0.0.0:4317+` is used.
-<2> The OTLP HTTP endpoint. If omitted, the default `+0.0.0.0:4318+` is used.
-<3> The TLS server side configuration. Defines paths to TLS certificates. If omitted, TLS is disabled.
+<2> The TLS server side configuration. Defines paths to TLS certificates. If omitted, TLS is disabled.
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:178]:
```suggestion
<2> The server-side TLS configuration. Defines paths to TLS certificates. If omitted, TLS is disabled.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 2]
```diff
@@ -172,8 +175,11 @@ The OTLP receiver ingests data using the OpenTelemetry protocol (OTLP).
...
+<3> Path to the TLS certificate to use by the server to verify a client certificate. (optional) This sets the ClientCAs and ClientAuth to RequireAndVerifyClientCert in the TLSConfig. Please refer to link:https://godoc.org/crypto/tls#Config[Golang TLS Config] for more information.
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:179]:
```suggestion
<3> The path to the TLS certificate at which the server verifies a client certificate. Optional: You can set the `ClientCAs` and `ClientAuth` to `RequireAndVerifyClientCert` in the `TLSConfig`. See link:https://godoc.org/crypto/tls#Config[Golang TLS Config] for more information.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 3]
```diff
@@ -172,8 +175,11 @@ The OTLP receiver ingests data using the OpenTelemetry protocol (OTLP).
...
+<4> Specifies the duration after which the certificate will be reloaded. If not set, it will never be reloaded. Accepts a duration string, valid time units are `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:180]: I wouldn't recommend using `duration` unless clearly specifying the  `duration of [something]`. Perhaps replace it `time interval` like in my suggestion below?
```suggestion
<4> Specifies the time interval at which the certificate is reloaded. If not set, the certificate is never reloaded. Accepts a string with valid units of time such as `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 4]
```diff
@@ -172,8 +175,11 @@ The OTLP receiver ingests data using the OpenTelemetry protocol (OTLP).
...
+<5> The OTLP HTTP endpoint. If omitted, the default `+0.0.0.0:4318+` is used.
+<6> The TLS server side configuration. See the `grpc` protocol configuration section for more details.
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:182]:
```suggestion
<6> The server-side TLS configuration. See the `grpc` protocol configuration section for more details.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 5]
```diff
@@ -308,7 +317,11 @@ The OTLP gRPC exporter exports data using the OpenTelemetry protocol (OTLP).
 ----
 <1> The OTLP gRPC endpoint. If the `+https://+` scheme is used, then client transport security is enabled and overrides the `insecure` setting in the `tls`.
 <2> The client side TLS configuration. Defines paths to TLS certificates.
-<3> Headers are sent for every RPC performed during an established connection.
+<3> Whether to disable client transport security. Disabled by default.
+<4> Whether to skip verifying the certificate or not. Disabled by default.
+<5> Specifies the duration after which the certificate will be reloaded. If not set, it will never be reloaded. Accepts a duration string, valid time units are `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:322]:
```suggestion
<5> Specifies the time interval at which the certificate is reloaded. If not set, the certificate is never reloaded. Accepts a string containing valid units of time such as `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 6]
```diff
@@ -308,7 +317,11 @@ The OTLP gRPC exporter exports data using the OpenTelemetry protocol (OTLP).
 ----
 <1> The OTLP gRPC endpoint. If the `+https://+` scheme is used, then client transport security is enabled and overrides the `insecure` setting in the `tls`.
 <2> The client side TLS configuration. Defines paths to TLS certificates.
-<3> Headers are sent for every RPC performed during an established connection.
+<3> Whether to disable client transport security. Disabled by default.
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:320]:
```suggestion
<3> Disables client transport security when set to `true`; `false` by default.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 7]
```diff
@@ -308,7 +317,11 @@ The OTLP gRPC exporter exports data using the OpenTelemetry protocol (OTLP).
...
+<4> Whether to skip verifying the certificate or not. Disabled by default.
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:321]:
```suggestion
<4> Skips verifying the certificate when set to `true`; `false` by default.
```

File: modules/distr-tracing-otel-config-collector.adoc (hunk 2 above)
Comment (pavolloffay) [reply modules/distr-tracing-otel-config-collector.adoc:179]: I wanted to say that this setting is optional

File: modules/distr-tracing-otel-config-collector.adoc [hunk 8]
```diff
@@ -308,7 +318,11 @@ The OTLP gRPC exporter exports data using the OpenTelemetry protocol (OTLP).
 ----
 <1> The OTLP gRPC endpoint. If the `+https://+` scheme is used, then client transport security is enabled and overrides the `insecure` setting in the `tls`.
 <2> The client side TLS configuration. Defines paths to TLS certificates.
-<3> Headers are sent for every RPC performed during an established connection.
+<3> Disables client transport security when set to `true`; `false` by default.
+<4> Skips verifying the certificate when set to `true`; `false` by default.
+<5> Specifies the time interval at which the certificate is reloaded. If not set, the certificate is never reloaded. Accepts a string containing valid units of time such as `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.
+<6> If set to a non-empty string, it will override the virtual host name of authority (e.g. :authority header field) in requests (typically used for testing).
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:324]: I'm confused about `<6>` and `<7>` now. Could you please check these two callouts in the codeblock?
```

Comment (max-cx) [reply modules/distr-tracing-otel-config-collector.adoc:324]:
```suggestion
<6> When set to a non-empty string, overrides the virtual host name of authority such as the authority header field in requests. Use this for testing.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 9]
```diff
@@ -297,7 +303,11 @@ The OTLP gRPC exporter exports data using the OpenTelemetry protocol (OTLP).
           ca_file: ca.pem
           cert_file: cert.pem
           key_file: key.pem
-        headers: <3>
+          insecure: false <3>
+          insecure_skip_verify: false <4>
+          reload_interval: 1h <5>
+          server_name_override <6>
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:309]:
```suggestion
          server_name_override: "<string>" <6>
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 10]
```diff
@@ -308,7 +318,11 @@ The OTLP gRPC exporter exports data using the OpenTelemetry protocol (OTLP).
 ----
 <1> The OTLP gRPC endpoint. If the `+https://+` scheme is used, then client transport security is enabled and overrides the `insecure` setting in the `tls`.
 <2> The client side TLS configuration. Defines paths to TLS certificates.
-<3> Headers are sent for every RPC performed during an established connection.
+<3> Disables client transport security when set to `true`; `false` by default.
+<4> Skips verifying the certificate when set to `true`; `false` by default.
+<5> Specifies the time interval at which the certificate is reloaded. If not set, the certificate is never reloaded. Accepts a string containing valid units of time such as `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.
+<6> When set to a non-empty string, overrides the virtual host name of authority such as the authority header field in requests. Use this for testing.
+<7> Headers are sent for every RPC performed during an established connection.
```
[the remark this answers is not in the document]
Comment (pavolloffay) [reply modules/distr-tracing-otel-config-collector.adoc:325]: remote procedure call.

I have changed it to request

**Review by max-cx — APPROVED** [review]
Peer/language review: LGTM.

File: modules/distr-tracing-otel-config-collector.adoc [hunk 11]
```diff
@@ -172,8 +175,11 @@ The OTLP receiver ingests data using the OpenTelemetry protocol (OTLP).
           receivers: [otlp]
 ----
 <1> The OTLP gRPC endpoint. If omitted, the default `+0.0.0.0:4317+` is used.
-<2> The OTLP HTTP endpoint. If omitted, the default `+0.0.0.0:4318+` is used.
-<3> The TLS server side configuration. Defines paths to TLS certificates. If omitted, TLS is disabled.
+<2> The server-side TLS configuration. Defines paths to TLS certificates. If omitted, TLS is disabled.
+<3> The path to the TLS certificate at which the server verifies a client certificate. This sets the `ClientCAs` and `ClientAuth` to `RequireAndVerifyClientCert` in the `TLSConfig`. See link:https://godoc.org/crypto/tls#Config[Golang TLS Config] for more information.
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:179]:
```suggestion
<3> The path to the TLS certificate at which the server verifies a client certificate. This sets the `ClientCAs` and `ClientAuth` to `RequireAndVerifyClientCert` in the `TLSConfig`. See the link:https://godoc.org/crypto/tls#Config[documentation for the `Config` of the Golang TLS package] for more information.
```

**Review by IshwarKanse — APPROVED** [review]

File: modules/distr-tracing-otel-config-collector.adoc (hunk 11 above)
Comment (max-cx) [reply modules/distr-tracing-otel-config-collector.adoc]: Resolved.

**Comment by max-cx** [comment]
Peer/language review is done.

File: modules/distr-tracing-otel-config-collector.adoc [hunk 12]
```diff
@@ -172,8 +175,11 @@ The OTLP receiver ingests data using the OpenTelemetry protocol (OTLP).
           receivers: [otlp]
 ----
 <1> The OTLP gRPC endpoint. If omitted, the default `+0.0.0.0:4317+` is used.
-<2> The OTLP HTTP endpoint. If omitted, the default `+0.0.0.0:4318+` is used.
-<3> The TLS server side configuration. Defines paths to TLS certificates. If omitted, TLS is disabled.
+<2> The server-side TLS configuration. Defines paths to TLS certificates. If omitted, TLS is disabled.
+<3> The path to the TLS certificate at which the server verifies a client certificate. This sets the `ClientCAs` and `ClientAuth` to `RequireAndVerifyClientCert` in the `TLSConfig`. For more information, see the link:https://godoc.org/crypto/tls#Config[documentation for the `Config` of the Golang TLS package].
+<4> Specifies the time interval at which the certificate is reloaded. If not set, the certificate is never reloaded. Accepts a string containing valid units of time such as `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.
```
Comment (snarayan-redhat) [inline modules/distr-tracing-otel-config-collector.adoc:180]:
```suggestion
<4> Specifies the time interval at which the certificate is reloaded. If the value is not set, the certificate is never reloaded. `reload_interval` accepts a string containing valid units of time such as `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 13]
```diff
@@ -172,8 +175,11 @@ The OTLP receiver ingests data using the OpenTelemetry protocol (OTLP).
...
+<5> The OTLP HTTP endpoint. If omitted, the default `+0.0.0.0:4318+` is used.
```
Comment (snarayan-redhat) [inline modules/distr-tracing-otel-config-collector.adoc:181]:
```suggestion
<5> The OTLP HTTP endpoint. The default value is `+0.0.0.0:4318+`.
```

File: modules/distr-tracing-otel-config-collector.adoc (hunk 12 above)
Comment (snarayan-redhat) [reply modules/distr-tracing-otel-config-collector.adoc:180]: Is this field mandatory? If yes, you can start the statement with `Required:`

File: modules/distr-tracing-otel-config-collector.adoc [hunk 14]
```diff
@@ -172,8 +175,11 @@ The OTLP receiver ingests data using the OpenTelemetry protocol (OTLP).
...
+<6> The server-side TLS configuration. See the `grpc` protocol configuration section for more details.
```
Comment (snarayan-redhat) [inline modules/distr-tracing-otel-config-collector.adoc:182]:
```suggestion
<6> The server-side TLS configuration. For more information, see `grpc` protocol configuration section.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 15]
```diff
@@ -308,7 +318,11 @@ The OTLP gRPC exporter exports data using the OpenTelemetry protocol (OTLP).
 ----
 <1> The OTLP gRPC endpoint. If the `+https://+` scheme is used, then client transport security is enabled and overrides the `insecure` setting in the `tls`.
 <2> The client side TLS configuration. Defines paths to TLS certificates.
-<3> Headers are sent for every RPC performed during an established connection.
+<3> Disables client transport security when set to `true`; `false` by default.
```
Comment (snarayan-redhat) [inline modules/distr-tracing-otel-config-collector.adoc:321]:
```suggestion
<3> Disables client transport security when set to `true`. The default value is `false`.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 16]
```diff
@@ -308,7 +318,11 @@ The OTLP gRPC exporter exports data using the OpenTelemetry protocol (OTLP).
...
+<4> Skips verifying the certificate when set to `true`; `false` by default.
```
Comment (snarayan-redhat) [inline modules/distr-tracing-otel-config-collector.adoc:322]:
```suggestion
<4> Skips verifying the certificate when set to `true`. The default value is `false`.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 17]
```diff
@@ -308,7 +318,11 @@ The OTLP gRPC exporter exports data using the OpenTelemetry protocol (OTLP).
...
+<5> Specifies the time interval at which the certificate is reloaded. If not set, the certificate is never reloaded. Accepts a string containing valid units of time such as `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.
```
Comment (snarayan-redhat) [inline modules/distr-tracing-otel-config-collector.adoc:323]: Same comments as above

File: modules/distr-tracing-otel-config-collector.adoc [hunk 18]
```diff
@@ -308,7 +318,11 @@ The OTLP gRPC exporter exports data using the OpenTelemetry protocol (OTLP).
...
+<6> When set to a non-empty string, overrides the virtual host name of authority such as the authority header field in requests. Use this for testing.
```
Comment (snarayan-redhat) [inline modules/distr-tracing-otel-config-collector.adoc:324]: Can we also add what this field is? That would mean whatever value the user gives, the corresponding result is expected. We would have to specify `non-empty` string.
Also is `authority header` an example? If not, can we have an actual example?

```suggestion
<6> Overrides the virtual host name of authority such as the authority header field in requests. You can use this for testing.
```

**Review by snarayan-redhat — COMMENTED** [review]
Nice work! Few suggestions.

File: modules/distr-tracing-otel-config-collector.adoc (hunk 12 above)
Comment (pavolloffay) [reply modules/distr-tracing-otel-config-collector.adoc:180]: it's not mandatory

File: modules/distr-tracing-otel-config-collector.adoc (hunk 16 above)
Comment (pavolloffay) [reply modules/distr-tracing-otel-config-collector.adoc:322]: The original text was suggested by @max-cx. There should be probably agreement how to write this correctly. 

For now I will accept the suggestion.

**Comment by pavolloffay** [comment]
@snarayan-redhat PR updated

**Comment by jab-rh** [comment]
@pavolloffay, looks like this doesn't cherry-pick cleanly. Can you reconcile this conflict and we can merge into the remaining branches? Thanks!

**Comment by max-cx** [comment]
@/jab-rh and @pavolloffay, I believe the reason is the parent PR (https://github.com/openshift/openshift-docs/pull/62410) has not been cherry-picked.

**Comment by max-cx** [comment]
@jab-rh and @pavolloffay, the parent PR (https://github.com/openshift/openshift-docs/pull/62410) has since been cherry-picked. This PR is ready for the cherry-picks.

**Comment by max-cx** [comment]
@jab-rh, please see
https://github.com/openshift/openshift-docs/pull/64719#issuecomment-1752957787
and
https://github.com/openshift/openshift-docs/pull/64719#issuecomment-1752957982
above.

**Pull request merged**